plugins/vsphere/cloudkeeper_plugin_vsphere/__init__.py

```diff
@@ -0,0 +1,121 @@
+from datetime import datetime
+
+import cloudkeeper.logging
+from cloudkeeper.args import ArgumentParser
+from cloudkeeper.baseplugin import BaseCollectorPlugin
+from cloudkeeper.baseresources import BaseResource
+
+from .vsphere_client import new_vsphere_client
+from .resources import VSphereCluster, VSphereInstance, VSphereDataCenter
+
+from pyVmomi import vim
+
+log = cloudkeeper.logging.getLogger("cloudkeeper." + __name__)
+
+
+class VSphereCollectorPlugin(BaseCollectorPlugin):
+    cloud = "vsphere"
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.vsphere_client = new_vsphere_client()
+
+    def get_cluster(self) -> VSphereCluster:
+        return VSphereCluster(ArgumentParser.args.vsphere_host, {})
+
+    def get_keymap_from_vmlist(self, list_vm) -> VSphereCluster:
+        keyMap = {}
+        for key in list_vm[0].availableField:
+            keyMap[key.key] = key.name
+
+        return keyMap
+
+    def get_custom_attributes(self, vm, keymap):
+        attr = {}
+        for value in vm.value:
+            attr[str(keymap[value.key])] = str(value.value)
+
+        return attr
+
+    def add_instances(self, parent: BaseResource) -> None:
+        content = self.vsphere_client.client.RetrieveContent()
+
+        container = content.rootFolder  # starting point to look into
+        view_type = [vim.VirtualMachine]  # object types to look for
+        recursive = True  # whether we should look into it recursively
+        container_view = content.viewManager.CreateContainerView(
+            container, view_type, recursive
+        )
+
+        VMs = container_view.view
```

Comment:
Variable names in python should be snake case. E.g. `vms` unless `VMs` is a reference to an uninstantiated list of classes.